### PR TITLE
Bump sdk and remove BigInt

### PR DESCRIPTION
**Pull request by sisuresh**
This PR bumps the sdk to a version that no longer has BigInt. 

1. Note that all BigInt references in the examples have been replaced with i128 (not u128) to match the built-in token contract in rs-soroban-env, but the i128 vs u128 decision is still being discussed here https://discord.com/channels/897514728459468821/1047281257312161832/1047281257312161832.
2. The liquidity pool contract now includes the num-integer crate so we can take the square root of an i128.
3. I added some overflow protection, but we need to go through the examples thoroughly and make sure all arithmetic is safe . Issue for this - https://github.com/stellar/soroban-examples/issues/178.

### CODE REVIEW

**Review by leighmcculloch — APPROVED** [review]
Looks good, but I don't think the manual overflow checks are necessary if we're just going to panic. See below for why.

File: liquidity_pool/src/lib.rs [hunk 1]
```diff
@@ -81,57 +82,56 @@ fn put_token_share(e: &Env, contract_id: BytesN<32>) {
...
 }
 
-fn transfer_b(e: &Env, to: Identifier, amount: BigInt) {
+fn transfer_b(e: &Env, to: Identifier, amount: i128) {
     transfer(&e, get_token_b(&e), to, amount);
 }
 
+fn safe_mult(a: i128, b: i128) -> i128 {
+    a.checked_mul(b).expect("no overflow")
+}
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply liquidity_pool/src/lib.rs:133]: Ah ok that's good to know. I'll undo this change. It looks like `overflow-checks` doesn't check divide by 0, so I think `checked_div` is still safer than `/`.

[the remark this answers is not in the document]
Comment (sisuresh) [reply liquidity_pool/src/lib.rs:133]: Oh wait does `/` map to `checked_div`?

[the remark this answers is not in the document]
Comment (leighmcculloch) [reply liquidity_pool/src/lib.rs:133]: Addition with overflow:
```rust
pub fn main() {
    let mut args = std::env::args();
    args.next();
    let a: u32 = args.next().unwrap().parse().unwrap();
    let b: u32 = args.next().unwrap().parse().unwrap();
    let c = a + b;
    println!("{}", c);
}
```
```console
❯ rustc -C overflow-checks=off add.rs
❯ ./main 4294967295 1
0
❯ rustc -C overflow-checks=on add.rs
❯ ./add 4294967295 1
thread 'main' panicked at 'attempt to add with overflow', add.rs:5:13
```

Division by zero:
```rust
pub fn main() {
    let mut args = std::env::args();
    args.next();
    let a: u32 = args.next().unwrap().parse().unwrap();
    let b: u32 = args.next().unwrap().parse().unwrap();
    let c = a / b;
    println!("{}", c);
}
```
```console
❯ rustc -C overflow-checks=on div.rs
❯ ./div 1 0
thread 'main' panicked at 'attempt to divide by zero', div.rs:6:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
✘ exit status 101                                                                                                                                                       
❯ rustc -C overflow-checks=off div.rs
❯ ./div 1 0
thread 'main' panicked at 'attempt to divide by zero', div.rs:6:13
```

Division by zero seems to be unrelated to overflow checks.

[the remark this answers is not in the document]
Comment (leighmcculloch) [reply liquidity_pool/src/lib.rs:133]: @graydon Is the understanding I posted above, that we don't need to do `checked_().expect()` correct?

[the remark this answers is not in the document]
Comment (sisuresh) [reply liquidity_pool/src/lib.rs]: I reverted the overflow commit. We can look over this and address issues as part of https://github.com/stellar/soroban-examples/issues/178.

**Review by leighmcculloch — APPROVED** [review]

File: liquidity_pool/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (graydon) [reply liquidity_pool/src/lib.rs]: @leighmcculloch correct, the `overflow-checks=true` setting causes the compiler to insert overflow checks on all overflowing arithmetic operations. You can still manually request checked, wrapping or saturating operations if you want them, the `overflow-checks` flag just switches the default operator interpretation from wrapping to checked-and-implicitly-unwrapping. 

Divide-by-zero is a different case: there's not really a simple / credibly-expected-by-anyone set of alternative behaviour to switch the default to (there are wrapping and saturating variants but they deal with over/underflow, not divide-by-zero). It's just always an error: by default rust injects a check-for-zero-and-panic, but even if you ask it not to do that your CPU will generate SIGFPE and/or webassembly will trap on zero. There's no unchecked-and-behaves-ok fast path. The fastest non-potentially-crashing behaviour you can do is `checked_div` and then substitute-some-value yourself, manually.

**Pull request merged**